hugr-core/src/ops/constant/custom.rs

```diff
@@ -90,6 +88,33 @@ pub trait CustomConst:
     fn get_type(&self) -> Type;
 }
 
+/// Prerequisite for `CustomConst`. Allows to declare a custom hash function, but the easiest
+/// options are either to `impl MaybeHash for ... {}` to declare "not hashable", or else
+/// to implement (or derive) [Hash].
+pub trait MaybeHash {
+    /// Hashes the value, if possible; else return `false` without mutating the `Hasher`.
+    /// This relates with [CustomConst::equal_consts] just like [Hash] with [Eq]:
+    /// * if `x.equal_consts(y)` ==> `x.maybe_hash(s)` behaves equivalently to `y.maybe_hash(s)`
+    /// * if `x.hash(s)` behaves differently from `y.hash(s)` ==> `x.equal_consts(y) == false`
+    ///
+    /// As with [Hash], these requirements can trivially be satisfied by either
+    /// * `equal_consts` always returning `false`, or
+    /// * `maybe_hash` always behaving the same (e.g. returning `false`, as it does by default)
+    ///
+    /// Note: this uses `dyn` rather than being parametrized by `<H: Hasher>` so that we can
+    /// still use `dyn CustomConst`.
```

Comment:
I.e. wee need the trait to be [object safe](https://doc.rust-lang.org/reference/items/traits.html#object-safety)

Reply:
Comment updated.